### DISCUSSION TITLE
How to properly fix "Unwanted System Libraries"?

### DISCUSSION THREAD

**Discussion by Southpaw1496** [Homebrew/discussions#3936] [Writing Formulae/Casks] [answered]
My formula currently fails the `test-bot` GitHub Action on Ubuntu due to the error "Error: linkage failed". After reading [this past discussion](https://github.com/orgs/Homebrew/discussions/2176), I found that the error was indeed being caused by Unwanted System Libraries, namely `/lib/x86_64-linux-gnu/libcrypto.so.1.1` and `/lib/x86_64-linux-gnu/libssl.so.1.1`. The discussion I found seems to be saying that I either need to depend on them properly through Homebrew or not use them at all, but neither Libcrypto or LibSSL seem to have Homebrew formulas (from a quick `brew search`, at any rate). If I can't remove them as dependencies, what's the best way to fix the error? Has the best practice changed since that discussion was written?

Pull Request: https://github.com/southpaw1496/homebrew-quilt/pull/2 ([GitHub Actions run](https://github.com/Southpaw1496/homebrew-quilt/actions/runs/3490198243/jobs/5842064479))

**Comment by SMillerDev** [comment]
Aren't those libraries from openSSL?

**Reply by Southpaw1496** [reply]
Indeed they are, I realised that after Googling what they were. I did try depending on OpenSSL, but [that failed too](https://github.com/Southpaw1496/homebrew-quilt/actions/runs/3492156855/jobs/5845622635). I'm now going to try and pin it at OpenSSL 1.1

**Reply by Southpaw1496** [reply]
Pinning it [didn't work](https://github.com/Southpaw1496/homebrew-quilt/actions/runs/3492298947/jobs/5845926645)

**Reply by carlocab** [reply]
Try adding
```ruby
depends_on "pkg-config" => :build
```

**Reply by Southpaw1496** [reply]
[This also didn't work](https://github.com/Southpaw1496/homebrew-quilt/actions/runs/3495264694/jobs/5851834438)

**Comment by carlocab** [comment, marked as answer]
Try
```ruby
depends_on "openssl@1.1"
```

**Reply by Southpaw1496** [reply]
Surprisingly, this didn't work.

[Relevant GitHub Actions run](https://github.com/Southpaw1496/homebrew-quilt/actions/runs/3492298947/jobs/5845926645)

**Reply by Southpaw1496** [reply]
I realised I accidentally added it as a build dependency instead of a regular one.
